### ISSUE TITLE
2.4.11: Definition of perimeter is not consistent

### ISSUE THREAD

**Issue by JAWS-test** [w3c/wcag#2249]
[Definition of perimeter](https://w3c.github.io/wcag/guidelines/22/#dfn-perimeter) is not consistent:

* there is a purely mathematical definition that uses perimeter (e.g. 2𝜋r for circle, 2h+2w for rectangle).  This is stated in the definition as applicable to the circle
* there is a technical definition that uses the actual pixels used (e.g. 2h+2w-4 - and anything for the circle, but never 2𝜋r and probably usually not 2𝜋r rounded up or down). This is stated in the definition as applying to the rectangle.
* It remains unclear which method should be used for which element (e.g. for a rectangle with rounded corners)

I would be in favor of using a consistent method. Since both methods are too difficult for complex figures, my suggestion would be that we only ever use the minimum bounding box.

Related: https://github.com/w3c/wcag/issues/1324, https://github.com/w3c/wcag/issues/2222

**Comment by patrickhlauke** [comment]
while it's nice to go for simplicity, oversimplifications like that lead to illogical requirements. "why does my outline/border need to be thicker just because my shape isn't rectangular? do people with disabilities have trouble recognising shapes that are not rectangles?"

**Comment by JAWS-test** [comment]
>  do people with disabilities have trouble recognising shapes that are not rectangles?"

No, not at all. Rather, I think it makes the test more difficult. But anyway: What is important to me is that perimeter is defined unambiguously and not in 2 different ways (mathematical and pixel-based).

**Comment by patrickhlauke** [comment]
oh for sure, agree that there should be one clear definition of what is meant by perimeter. just throwing in that i'd hate to see the one definition being used be the one that's "simple" but illogical

**Comment by alastc** [comment]
So the [latest proposal](https://docs.google.com/document/d/1TI_DsJjfg9RW_A9C1XfqgtfYqNMfnxeCz6lxjpOQ9rk/edit#) (in the survey for discussion next week) doesn't use that definition. That's a proposal + discussion + test cases.

I'm not sure if the question is still relevant if that proposal is approved? The first method is to use an indicator that 'encompasses' the component, but (with an assumption you can't have less than a 1px thickness) it doesn't try to set a minimum area, it just has to go around (i.e. not inside) the component.

That should be straightforward to calculate. However, if it doesn't meet that, you'd need to fall-back to the exception with the 4px * shortest-side approach.

**Comment by patrickhlauke** [comment]
so just to be crystal, the actual perimeter of a circular or triangular or whatever control would "encompass" the control, right?

**Comment by alastc** [comment]
> the actual perimeter of a circular or triangular or whatever control would "encompass" the control

Yes. It would need to be the pixel outside of the component, rather than on the component, but see the doc for a discussion of interpreting UIC for size & active descendants.

**Comment by patrickhlauke** [comment]
perfect, thanks @alastc
